### DISCUSSION TITLE
Low mAP and mAR for centred instance model

### DISCUSSION THREAD

**Discussion by chrisch12138** [talmolab/sleap#1479] [Help!]
Hi,

For my project, I labeled 950 frames from 14 five-hour recordings, an example of the recording is attached below. I tried top-down method, and the first controid model seems decent, but the second model (centred instance) does not seem to have a good performance (very low mAP and mAR). The model parameters and metrics are shown attached. I'm wondering if there's any change of the parameter I could try to make the second model better? Thanks!!
![recording example](https://github.com/talmolab/sleap/assets/95929367/d136e6f3-daf6-4db8-8168-0f339f2a4411)
parameters for centroid
![centroid n=950](https://github.com/talmolab/sleap/assets/95929367/09f6523b-9d0b-430a-8c37-e7bb1fa60faf)
Metrics for centroid model
![centroid metrics](https://github.com/talmolab/sleap/assets/95929367/f9147f90-15c6-41f0-a2a0-7d0c3e136abe)
parameters for centred instance 
![centered instance n=950](https://github.com/talmolab/sleap/assets/95929367/41eb025e-4324-427b-8968-4458c903f969)
metrics for centred instance
![centered instance metrics](https://github.com/talmolab/sleap/assets/95929367/06da501b-b64b-49ca-941a-843858f0bab5)
screenshot of the training pipeline
![pipeline](https://github.com/talmolab/sleap/assets/95929367/2abeaa01-4c0b-4e73-b0b9-3a90398c0efe)

**Comment by talmo** [comment]
Hi @chrisch12138,

Thanks for reaching out!

I'd definitely recommend increasing your max stride on both models to at least 16, and ideally 32 for the centered instance model.

If you turned those down because of memory issues, you can try decreasing the filter rate to 1.5.

The other parameters look pretty reasonable. Try it out with the increased max stride and let us know how it goes!

Cheers,

Talmo

**Reply by chrisch12138** [reply]
Hi Talmo,

Thanks for the suggestions. I adjusted the parameters you suggested, and I have some outputs. Looks like they improved! That's terrific! However, I kept running into some memory caller issues (not failure, as it stresses), so I turned down the batch size to 3, but the caller still appeared. Should I do anything to reduce memory issues for better model performance? Overall, is there anything else I can try to improve the model performance to achieve the levels provided from your examples? (From the examples on googleColab, it looks like the mAP and mAR were around 0.8..)

![memory issue output](https://github.com/talmolab/sleap/assets/95929367/16c53a2a-5da9-4c00-b205-4328e953c933)

Centroid performance improved
![0908 centroid](https://github.com/talmolab/sleap/assets/95929367/0351e434-8610-4506-a318-b8aa73cf7bbb)

Centered instance performance improved
![0908centered instance](https://github.com/talmolab/sleap/assets/95929367/a2bf7d2e-db37-4c91-b6a0-a0b4d410828a)

Thanks!

Chris

**Reply by talmo** [reply]
Hi @chrisch12138,

Good to see that improved things!

What was the final configuration you landed on for the centroid model?

For centered instance model, what I'd recommend is focusing on the outliers now and label a few examples (20-30) of suggested frames with low prediction score. I think you're now in the final stages of fine tuning and optimizing the model!

Memory warnings: which model does this occur during? Both?

Not much you can do since it's likely that it just takes a minimum amount of GPU memory to train on your image size. You can decrease the filters and filters rate, but I wouldn't advise it unless you absolutely have to since it'll hurt the accuracy.

Cheers,

Talmo

**Reply by chrisch12138** [reply]
Hi @talmo,

Thanks for the suggestions! The final configuration I used for the centroid model is shown in the screenshot. I tried to turn down the batch size to 3 but the memory issue message still appeared, so I eventually turned it back to 4. And yes, the memory warnings only occurred during centroid model training.

I labeled another 50 frames (the relatively more the merrier right? :) that have prediction scores around or less than 0.5, and it looks like the performance of both models improved. Does this mean I should keep labeling more frames for better results? When will the plateau approximately be reached? Right now the labeled frames are 1000, but that's for 14 videos of 540 000 frames each.

Centroid configuration
![centroid configuration n=1000 0912](https://github.com/talmolab/sleap/assets/95929367/f7053331-cd64-424b-90dd-928a6a1c187c)

Centroid model metrics
![0912centroid](https://github.com/talmolab/sleap/assets/95929367/13aafe2c-bae9-41d5-8794-c36c9d8cb201)

Centred instance model metrics
![0912centered instance](https://github.com/talmolab/sleap/assets/95929367/4b8c6c82-2aa3-49fc-8767-d484e176b201)

Thanks,

Chris
